core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java

```diff
@@ -101,6 +101,9 @@ public void validateInstanceGroup(Stack stack, String instanceGroupName) {
         if (instanceGroup == null) {
             throw new BadRequestException(format("Stack '%s' does not have an instanceGroup named '%s'.", stack.getName(), instanceGroupName));
         }
+        if (!instanceGroup.getScalingMode().isManuallyScalable()) {
+            throw new BadRequestException(format("Instance group '%s' in stack '%s' is not enabled to scale", instanceGroupName, stack.getName()));
+        }
```

Comment:
this is very similar to the previous one, could you extract this logic into a class and reuse it?

Reply:
Thanks, done!